shared/actions/profile/proofs.js

```diff
@@ -0,0 +1,284 @@
+}
+
+function _updateErrorText (errorText: ?string, errorCode: ?number): UpdateErrorText {
+  return {type: Constants.updateErrorText, payload: {errorText, errorCode}}
+}
+
+function _updateSigID (sigID: ?SigID): UpdateSigID {
+  return {type: Constants.updateSigID, payload: {sigID}}
+}
+
+function checkProof (): CheckProof {
+  return {type: Constants.checkProof, payload: undefined}
+}
+
+function * _checkProof (action: CheckProof): SagaGenerator<any, any> {
+  const sigID = yield select(state => state.profile.sigID)
+  if (!sigID) {
+    return
+  }
+
+  yield put(_updateErrorText(null))
+
+  try {
+    yield put(_waitingForResponse(true))
+    // $ForceType
+    const {found, status} = yield call(proveCheckProofRpcPromise, {param: {sigID}})
+    yield put(_waitingForResponse(false))
+    // this enum value is the divider between soft and hard errors
+    if (!found && status >= ProveCommonProofStatus.baseHardError) {
+      yield put(_updateErrorText("We couldn't find your proof. Please retry!"))
+    } else {
+      yield put(_updateProofStatus(found, status))
+      yield put(navigateTo([{path: 'ConfirmOrPending'}], profileTab))
+    }
+  } catch (error) {
+    yield put(_waitingForResponse(false))
+    console.warn('Error getting proof update')
+    yield put(_updateErrorText("We couldn't verify your proof. Please retry!"))
+  }
+}
+
+function * _addProof (action: AddProof): SagaGenerator<any, any> {
+  yield put(_updatePlatform(action.payload.platform))
+  yield put(_updateErrorText())
+
+  // Special cases
+  switch (action.payload.platform) {
+    case 'dnsOrGenericWebSite':
+      yield put(_askTextOrDNS())
+      break
+    case 'btc':
+      yield yield put(_registerBTC())
+      break
+    // flow needs this for some reason
+    case 'http':
+    case 'https':
+    case 'twitter':
+    case 'facebook':
+    case 'reddit':
+    case 'github':
+    case 'coinbase':
+    case 'hackernews':
+    case 'dns':
+      yield call(_addServiceProof, action.payload.platform)
+      break
+    case 'pgp':
+      yield put(routeAppend(['pgp', 'choice']))
+  }
+}
+
+function * _addServiceProof (service: ProvablePlatformsType): SagaGenerator<any, any> {
+  let _promptUsernameResponse: ?Object = null
+  let _outputInstructionsResponse: ?Object = null
+
+  const channelConfig = singleFixedChannelConfig([
+    'keybase.1.proveUi.promptUsername',
+    'keybase.1.proveUi.outputInstructions',
+    'keybase.1.proveUi.promptOverwrite',
+    'keybase.1.proveUi.outputPrechecks',
+    'keybase.1.proveUi.preProofWarning',
+    'keybase.1.proveUi.okToCheck',
+    'keybase.1.proveUi.displayRecheckWarning',
+    'finished',
+  ])
+
+  yield put(_updateSigID(null))
+
+  const proveStartProofChanMap: ChannelMap<any> = proveStartProofRpcChannelMap(channelConfig, {
+    param: {
+      auto: false,
+      force: true,
+      promptPosted: false,
+      service,
+      username: '',
+    },
+  })
+
+  while (true) {
```

Comment:
emulating how the old code worked. This isn't a normal linear flow. Some steps can be skipped. Instead of rethinking this logic and trying to make sure it all worked i'm just emulating how the old incomingMap worked.